ci-operator/step-registry/sippy-e2e/sippy-e2e-commands.sh

```diff
@@ -0,0 +1,316 @@
+    env:
+    - name: GCS_SA_JSON_PATH
+      value: /tmp/secrets/openshift-ci-data-analysis-ro
+    volumeMounts:
+    - mountPath: /tmp/secrets
+      name: gcs-cred
+      readOnly: true
+  imagePullSecrets:
+  - name: regcred
+  volumes:
+    - name: gcs-cred
+      secret:
+        secretName: gcs-cred
+  dnsPolicy: ClusterFirst
+  restartPolicy: Always
+  schedulerName: default-scheduler
+  securityContext: {}
+  terminationGracePeriodSeconds: 30
+END
+
+date
+oc -n postgres get po sippy-load
+oc -n postgres describe po sippy-load
+
+is_ready=0
+for i in `seq 1 20`; do
+  echo "${i} Sleeping 30s ..."
+  sleep 30
+  c=$(oc -n postgres logs sippy-load|tail -1|grep IM_DONE|wc -l)
+  if [ $c -eq 1 ]; then
+    echo "sippy load is done."
+    is_ready=1
+    break
+  fi
+  echo "sippy loader pod not done yet ..."
+done
+
+if [ $is_ready -eq 0 ]; then
+  echo "sippy loading never finished on time."
+  exit 1
+fi
+
+# Check to see if we got an error on sippy loader.
+c=$(oc -n postgres logs sippy-load|tail -2|grep "error running command"|wc -l)
+if [ $c -gt 0 ]; then
+  echo "sippy loading got an error."
+  oc -n postgres logs sippy-load
+  exit 1
+fi
+
+date
+oc -n postgres logs sippy-load | tail -10
+
+# Launch the sippy api server pod.
+cat << END | oc apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sippy-server
+  namespace: postgres
+  labels:
+    app: sippy-server
+spec:
+  containers:
+  - name: sippy-server
+    image: ${SIPPY_IMAGE}
+    imagePullPolicy: Always
+    ports:
+    - name: www
+      containerPort: 8080
+      protocol: TCP
+    - name: metrics
+      containerPort: 12112
+      protocol: TCP
+    resources:
+      limits:
+        memory: 2G
+    terminationMessagePath: /dev/termination-log
+    terminationMessagePolicy: File
+    command:
+    - /bin/sippy
+    args:
+    - --server
+    - --listen
+    - ":8080"
+    - --listen-metrics
+    -  ":12112"
+    - --local-data
+    -  /opt/sippy-testdata
+    - --database-dsn=postgresql://postgres:password@postgres.postgres.svc.cluster.local:5432/postgres
+    - --log-level
+    - debug
+    - --mode
+    - ocp
+  imagePullSecrets:
+  - name: regcred
+  dnsPolicy: ClusterFirst
+  restartPolicy: Always
+  schedulerName: default-scheduler
+  securityContext: {}
+  terminationGracePeriodSeconds: 30
+END
+
+is_ready=0
+for i in `seq 1 20`; do
+  echo "${i} Sleeping 30s ..."
+  sleep 30
+  c=$(oc -n postgres logs sippy-server|tail -1|grep "Refresh complete"|wc -l)
+  if [ $c -eq 1 ]; then
+    echo "sippy server is ready."
+    is_ready=1
+    break
+  fi
+  echo "sippy server pod not ready yet ..."
+done
+if [ $is_ready -eq 0 ]; then
+  echo "sippy server became ready."
+  exit 1
+fi
+
+oc -n postgres get pod -o wide
+oc -n postgres logs sippy-server
+
+echo "Exposing services and routes for the sippy api server ..."
+set -x
+oc -n postgres expose pod/sippy-server
+oc -n postgres expose svc sippy-server
```

Comment:
Do we need to expose both pod and svc?

Reply:
yes, exposing the pod creates the service.  Exposing the service creates the route.